### PR TITLE
Legacy compatibility

### PR DESCRIPTION
**Pull request by ianwremmel**
Review commit-by-commit

[UD-5858](https://urbandoor.atlassian.net/browse/UD-5858)
[UD-5859](https://urbandoor.atlassian.net/browse/UD-5859)
[UD-5860](https://urbandoor.atlassian.net/browse/UD-5860)
[UD-6252](https://urbandoor.atlassian.net/browse/UD-6252)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ dependencies) onto your page.
 > `window.fetch()`. The easiest way to do so is with polyfill.io
 >
 > ```html
-> <script src="https://cdn.polyfill.io/v2/polyfill.min.js"></script>
+> <script src="https://cdn.polyfill.io/v2/polyfill.min.js?features=fetch|gate"></script>
```
Comment (NathanDeVuono) [inline README.md]: 👏

File: src/lib/polyfills.js [hunk 2]
```diff
@@ -79,3 +79,16 @@ export function getAttributeNames(el) {
...
+ * @param {string} haystack
+ * @param {string} needle
+ * @returns {boolean}
+ */
+export function startsWith(haystack, needle) {
+  if (typeof haystack.startsWith === 'function') {
+    return haystack.startsWith(needle);
+  }
+
+  return haystack.indexOf(needle) === 0;
```
Comment (NathanDeVuono) [inline src/lib/polyfills.js]: Lul

**Comment by ianwremmel** [comment]
> Is there a way to setup linting to prevent those instance methods from being used directly here?

I thought so, but it turns out, eslint-plugin-compat isn't as magical as I'd hoped. Tests will fail (hence the addition of Edge, IE, and Safari). Some of the functions will be caught by typescript  (which is why I removed the type augmentations).

Actually, let me see if I can mark the remaining ones as `never` in the typescript d files.

**Comment by ianwremmel** [comment]
> Actually, let me see if I can mark the remaining ones as `never` in the typescript d files.

Won't work since `never` isn't part of the set of allowed types for e.g. `Element.closest`.

**Comment by ianwremmel** [comment]
I made a note in the CONTRIBUTING pr about putting something in the readme

**Comment by ianwremmel** [comment]
Note: I'm forcing this merge through because the tests passed, but the webhooks got out of sync.

**Comment by ianwremmel** [comment]
https://circleci.com/workflow-run/a1a22bfa-ed59-4997-a0c0-8c32130db13d

**Pull request merged**